internal/controller/operatorconfigmap_controller.go

```diff
@@ -1075,3 +1089,68 @@ func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error)
 	}
 	return tolerations, nil
 }
+
+func (c *OperatorConfigMapReconciler) removeNoobaa() error {
+	noobaa := &nbv1.NooBaa{}
+	noobaa.Name = "remote-noobaa"
+	noobaa.Namespace = c.OperatorNamespace
+
+	if err := c.get(noobaa); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get remote noobaa: %v", err)
+	} else if noobaa.UID != "" && noobaa.GetDeletionTimestamp().IsZero() {
+		index := slices.IndexFunc(
+			noobaa.GetOwnerReferences(), func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "StorageClient"
+			},
+		)
+		if index == -1 {
+			if err := c.delete(noobaa); err != nil {
+				return fmt.Errorf("failed to delete remote noobaa: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) removeNoobaaOperator() error {
+
+	nb := &metav1.PartialObjectMetadataList{}
+	nb.SetGroupVersionKind(nbv1.SchemeGroupVersion.WithKind("NooBaa"))
+	if err := c.list(nb); err != nil {
+		return fmt.Errorf("failed to list noobaa: %v", err)
+	}
+	if len(nb.Items) != 0 {
+		c.log.Info("noobaa cr still present on cluster, skipping removal of noobaa csv and subscription")
+		return nil
+	}
+
+	noobaaSubscription, err := c.getSubscriptionByPackageName("noobaa-operator")
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if noobaaSubscription.UID != "" {
+		index := slices.IndexFunc(
+			noobaaSubscription.OwnerReferences,
+			func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "Subscription" && ref.Name == "odf-operator"
+			},
+		)
+		if index != -1 {
+			c.log.Info("noobaa operator subscription is owned by odf-operator, skipping removal")
+			return nil
+		}
+	}
+	csv := &opv1a1.ClusterServiceVersion{}
+	csv.Name = noobaaSubscription.Status.InstalledCSV
+	csv.Namespace = c.OperatorNamespace
+	if csv.Name != "" {
+		if err := c.delete(csv); client.IgnoreNotFound(err) != nil {
```

Comment:
Not checking if CSV is owned by odf as we are already checking if subscription is owned by odf or not.
and we get CSV from Subscription

Reply:
Yes this is acceptable